packages/server/billing/helpers/terminateSubscription.ts

```diff
@@ -32,7 +32,7 @@ const terminateSubscription = async (orgId: string) => {
     try {
       await manager.deleteSubscription(stripeSubscriptionId)
     } catch (e) {
-      // noop
+      console.error(`cannot delete subscription ${stripeSubscriptionId}`, e)
```

Comment:
+1 Reading the comment in l28 it appears as if we intentionally do not track errors here. Are you sure this is necessary?

Reply:
I would highly suspect that it never did what it was supposed to do... the issue was hidden as we didn't have that many self upgrades in the past, let alone the even rare cases where the payment failed.

As a quick test, I commented out this part as the comment seemed to indicate, i.e., it wouldn't harm if the deletion failed or not even performed, as Stripe is configured to do so automatically. Well, not the case here. If the customer failed to perform the first payment and just did nothing, this is gonna happen:
1. A failed payment notification was sent
2. The invoice stays in PENDING state
3. The org still stays in the Team tier, i.e., the customer is enjoying the Team tier for free

<img width="1056" alt="Screenshot 2023-04-25 at 11 47 29" src="https://user-images.githubusercontent.com/1879975/234163244-84984807-0d40-49d4-97a3-feae1b281bde.png">

Looking at Stripe side, after 24 hours, this is going to happen:
1. The invoice is void
2. The payment was cancelled 
3. The subscription was incomplete
But again, none of those prevent us from giving the customers Team tier for free, from the app side.

<img width="1062" alt="Screenshot 2023-04-25 at 11 48 02" src="https://user-images.githubusercontent.com/1879975/234163524-ce1e64cc-cb37-4fa6-9986-1bb5cffa5cef.png">

So, all in all, we definitely need that subscription deletion to finish and succeed, in order for the app to cancel that upgrade. Therefore, we should capture the error.

Reply:
Thanks for investigating this and removing the comment!